Added Registration Page along with its Mongo Schemas and Registration Endpoint

shreyash3087 opened the pull request:
**Description:**

This PR adds the registration page,mongo schemas reqiured for it,and the registration's endpoint to connect to mongodb database.

Note: This PR doesnt focus on the frontend design,the page will be updated after the design for the registration page is completed.

**Video:**


https://github.com/user-attachments/assets/bff6b07c-9b16-4734-8e60-305b3ff52f03


**Additional Context:**

N/A

sanjay-kv commented:
Just wondering if i want to do analysis or create powerbi dashboard , what form the output will be csv or any API endpoints. 

I know you using MongoDB, will it allow to convert database to csv format?
or fetching data realtime and make an analysis page? @shreyash3087

sanjay-kv commented on pages/registration.js:
```diff
@@ -0,0 +1,246 @@
+            <InputField label="Discord Username" name="discordUsername" handleChange={handleInputChange} error={errors.discordUsername} />
+            <InputField label="Country" name="country" handleChange={handleInputChange} error={errors.country} />
+            <InputField label="Experience as CA" name="experienceAsCA" handleChange={handleInputChange} error={errors.experienceAsCA} />
+            <TextAreaField label="Reason" name="reason" handleChange={handleInputChange} error={errors.reason} />
```
We will be adding few more input fields here, I will confirm you by tomorrow morning.

shreyash3087 replied:
@sanjay-kv Sure

sanjay-kv reviewed (commented):
i will add more input field to ask by tomorrow.

shreyash3087 commented:
> m the output will be csv or



> Just wondering if i want to do analysis or create powerbi dashboard , what form the output will be csv or any API endpoints.
> 
> I know you using MongoDB, will it allow to convert database to csv format? or fetching data realtime and make an analysis page? @shreyash3087

Yes, the datbase can be converted to csv format,and yes we can make an analysis page as well, the data can be fetched directly from mongodb, it will return a json file with all the data from a particular collection which can be used to make admin dashboards for checking the no. of registrations and how many contributors/ca/admins have joined the program along with all the stats

Hemu21 commented:
> Just wondering if i want to do analysis or create powerbi dashboard , what form the output will be csv or any API endpoints. 
> 
> I know you using MongoDB, will it allow to convert database to csv format?
> or fetching data realtime and make an analysis page? @shreyash3087 

It allow to get the data in csv format. If needed we can make a admin dashboard type thing for the team to analyse the data as we make it private so no problem.

Hemu21 commented:
@sanjay-kv  we need to create the mongodb database for the website I am thinking if we create with gssoc mail or gssoc github account it would be great. Why because that data will be there forever until or unless we delete it.

sanjay-kv commented:
![image](https://github.com/user-attachments/assets/b9eeabf0-c3a6-4534-bfe6-b6055ab352d8)

https://github.com/GSSoC24/being-an-GSSoc24/issues/117

@shreyash3087 , I made some changes can you update the field . so there wont be any changes on it. 

@Hemu21 @MastanSayyad have a review on updated issue

shreyash3087 commented:
@sanjay-kv @MastanSayyad @Hemu21 The Fields are Updated

Added the country code based phone number selection and instead of just having indian states i have added country based state options in the input fields

**Note:** For now,as discussed with @MastanSayyad I have commented the option to choose Contributor,Mentor and Project Admin Role and have just kept the option to choose CA as the registrations for the other three will get open later ( to open the registrations for others,we just have to uncomment those options - 3 lines of code)


https://github.com/user-attachments/assets/1fdb6f3f-97db-4f53-bc35-8dc1387a5ddc

Hemu21 commented:
@shreyash3087  above the return function add if condition and condition is true and return something like coming soon. As of now we doesn't announced the program. If registration is open then people may think gssoc registration is open like that.

shreyash3087 commented:
@Hemu21 @MastanSayyad I guess we can merge it now

aditya-bhaumik commented:
@shreyash3087 why is there direct registration successful there is a second page also for mentors and project admins
